apis/v1alpha1/shared_types.go

```diff
@@ -211,6 +211,19 @@ type RouteStatus struct {
 	//
 	// +kubebuilder:validation:MaxItems=100
 	Gateways []RouteGatewayStatus `json:"gateways"`
```

Comment:
Should `Gateways` still exist? It seems redundant.

Reply:
The big difference is that this list of Gateways includes all the conditions/status for the associated Gateways. In this PR I'm proposing adding a simpler `[]string` for the purpose of populating kubectl output, but the full status/conditions would still be valuable.

Reply:
@robscott does kubectl provide a way to iterate through the gateway ref list and output gateways of "Admitted=True"?

Reply:
Even if we were able to get a fix into Kubernetes 1.22, the kubectl experience would be quite poor for anyone on an older k8s version since the current implementation always returns the first element in the matching list.

Reply:
So this is the names of the gateways that are included in the `Gateways` array and have an `Admitted` status of true? If so, it's redundant but I can see your point here.

The doc should specify the semantic described here. I don't think that the intention is to invent new semantics for this field; we just want to make existing semantics consumable by kubectl. We should avoid the word "bound" since that's not a term of art in the API.